tests/test_dataset_generator.py

```diff
@@ -91,8 +224,168 @@ def test_generate_custom_dataset(self):
+                        "August 19 2008 16:53:49", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "March 13 2010 17:18:44",
+                    datetime.strptime(
+                        "March 13 2010 17:18:44", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "March 11 2016 15:15:39",
+                    datetime.strptime(
+                        "March 11 2016 15:15:39", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "September 27 2018 18:24:03",
+                    datetime.strptime(
+                        "September 27 2018 18:24:03", self.date_format_list[0]
+                    ),
+                ],
+            ]
+        )
+
+        actual = dataset_generator.get_ordered_column(data, "datetime", "ascending")
+
+        np.testing.assert_array_equal(actual, expected[:, 0])
+
+    def test_get_ordered_column_datetime_descending(self):
+        data = datetime_generator.random_datetimes(
+            rng=self.rng, start_date=self.start_date, end_date=self.end_date, num_rows=5
+        )
+
+        expected = np.array(
+            [
+                [
+                    "September 27 2018 18:24:03",
+                    datetime.strptime(
+                        "September 27 2018 18:24:03", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "March 11 2016 15:15:39",
+                    datetime.strptime(
+                        "March 11 2016 15:15:39", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "March 13 2010 17:18:44",
+                    datetime.strptime(
+                        "March 13 2010 17:18:44", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "August 19 2008 16:53:49",
+                    datetime.strptime(
+                        "August 19 2008 16:53:49", self.date_format_list[0]
+                    ),
+                ],
+                [
+                    "October 02 2006 22:34:32",
+                    datetime.strptime(
+                        "October 02 2006 22:34:32", self.date_format_list[0]
+                    ),
+                ],
+            ]
+        )
+
+        actual = dataset_generator.get_ordered_column(data, "datetime", "descending")
+
+        np.testing.assert_array_equal(actual, expected[:, 0])
+
+    def test_get_ordered_column_custom_datetime_ascending(self):
+        custom_date_format = ["%Y %m %d", "%B %d %Y %H:%M:%S"]
+        data = datetime_generator.random_datetimes(
+            rng=self.rng,
+            date_format_list=custom_date_format,
+            start_date=self.start_date,
+            end_date=self.end_date,
+            num_rows=5,
+        )
+
+        expected = np.array(
```

Comment:
it looks like the other tests in `TestGetOrderedColumn` are trying to test `get_ordered_column` which returns an array of strings... but the expected values defined in the test aren't arrays of strings...

Reply:
the other tests have the expected of `[:,0]` here but not this test when the test are only changing from `ascending` to `descending`? Do we need `datetime.strptime("2018 09 27", custom_date_format[0]),` values in the `expected` array defined on 340?